src/main/java/org/gridsuite/modification/server/service/NetworkModificationService.java

```diff
@@ -220,4 +277,123 @@ private void saveModifications(NetworkStoreListener listener) {
+            .setName(loadCreationInfos.getEquipmentName())
+            .setLoadType(loadCreationInfos.getLoadType())
+            .setBus(bus.getId())
+            .setConnectableBus(bus.getId())
+            .setP0(loadCreationInfos.getActivePower())
+            .setQ0(loadCreationInfos.getReactivePower())
+            .add();
+    }
+
+    public Flux<EquipmenModificationInfos> createLoad(UUID networkUuid, UUID groupUuid, LoadCreationInfos loadCreationInfos) {
+        return assertLoadCreationInfosNotEmpty(loadCreationInfos).thenMany(
+                getNetwork(networkUuid).flatMapIterable(network -> {
+                    NetworkStoreListener listener = NetworkStoreListener.create(network, groupUuid, modificationRepository);
+                    ReporterModel reporter = new ReporterModel("NetworkModification", "Network modification");
+                    Reporter subReporter = reporter.createSubReporter("LoadCreation", "Load creation");
+
+                    return doAction(listener, () -> {
+                        // create the load in the network
+                        VoltageLevel voltageLevel = network.getVoltageLevel(loadCreationInfos.getVoltageLevelId());
+                        if (voltageLevel == null) {
+                            throw new NetworkModificationException(VOLTAGE_LEVEL_NOT_FOUND, loadCreationInfos.getVoltageLevelId());
+                        }
+                        if (voltageLevel.getTopologyKind() == TopologyKind.NODE_BREAKER) {
+                            createLoadInNodeBreaker(voltageLevel, loadCreationInfos);
+                        } else {
+                            createLoadInBusBreaker(voltageLevel, loadCreationInfos);
+                        }
+                        subReporter.report(Report.builder()
+                            .withKey("loadCreated")
+                            .withDefaultMessage("New load with id=${id} and name=${name} created")
+                            .withValue("id", loadCreationInfos.getEquipmentId())
+                            .withValue("name", loadCreationInfos.getEquipmentName())
+                            .withSeverity(new TypedValue("LOAD_CREATION_INFO", TypedValue.INFO_LOGLEVEL))
+                            .build());
+
+                        // add the load creation entity to the listener
+                        listener.storeLoadCreation(loadCreationInfos);
+                    }, CREATE_LOAD_ERROR, networkUuid, reporter, subReporter);
+                }));
+    }
+
+    private Mono<Void> assertLoadCreationInfosNotEmpty(LoadCreationInfos loadCreationInfos) {
+        return loadCreationInfos == null ? Mono.error(new NetworkModificationException(CREATE_LOAD_ERROR, "Missing required attributes to create the load")) : Mono.empty();
+    }
+
+    private void sendReport(UUID networkUuid, ReporterModel reporter) {
+        AtomicReference<Long> startTime = new AtomicReference<>(System.nanoTime());
```

Comment:
What is the purpose of `AtomicReference ?`

Reply:
Removed